src/input.rs

```diff
@@ -110,11 +113,35 @@ impl MidiInputConnection {
 #[derive(Resource)]
 pub struct MidiData {
     pub stamp: u64,
-    pub message: MidiMessage,
+    pub message: OwnedLiveEvent,
 }
 
 impl bevy::prelude::Event for MidiData {}
 
+impl MidiData {
+    /// Return `true` iff the underlying message represents a MIDI note on event.
+    pub fn is_note_on(&self) -> bool {
+        matches!(
+            self.message,
+            OwnedLiveEvent::Midi {
+                message: MidiMessage::NoteOn { .. },
+                ..
+            }
+        )
+    }
+
+    /// Return `true` iff the underlying message represents a MIDI note off event.
+    pub fn is_note_off(&self) -> bool {
```

Comment:
Would you agree that the logic for checking if NoteOn and Vel 0 should be made here?

Reply:
I actually wouldn't put it there, just because I think it's not intuitive for the `is_note_off` to not match the discriminant enum `NoteOn`. 

We could an option to this function, like `treat_vel0_as_off`, to make it less ambiguous, but my preference would be to make convenience class that actually transforms the message, so the user can call the transformer on every message and then interpret each message at face value. 

I'd also handle this in a separate PR, just so that this one doesn't get too large.